src/Symfony/Component/OptionsResolver/OptionsResolver.php

```diff
@@ -858,37 +867,32 @@ private function verifyArrayType($type, array $value, array &$invalidTypes, $lev
     {
         $type = substr($type, 0, -2);
 
-        $suffix = '[]';
-        while (\strlen($suffix) <= $level * 2) {
-            $suffix .= '[]';
-        }
-
         if ('[]' === substr($type, -2)) {
             $success = true;
             foreach ($value as $item) {
                 if (!\is_array($item)) {
-                    $invalidTypes[$this->formatTypeOf($item, null).$suffix] = true;
+                    $invalidTypes[$this->formatTypeOf($item, null)] = true;
 
-                    return false;
-                }
-
-                if (!$this->verifyArrayType($type, $item, $invalidTypes, $level + 1)) {
+                    $success = false;
+                } elseif (!$this->verifyArrayType($type, $item, $invalidTypes, $level + 1)) {
                     $success = false;
                 }
             }
 
             return $success;
         }
 
+        $valid = true;
```

Comment:
this should be reverted

Reply:
please, check tests `testNestedArrayException1`, `testNestedArrayException2` and `testResolveFailsIfTypedArrayContainsInvalidTypes` and confirm that after reverting it's okay to have error message with only one incorrect type like `...but one of the elements is of type "stdClass"` instead of `...but one of the elements is of type "stdClass|array|DateTime"`

Reply:
Ah, got it, we want to populate `$invalidTypes` with more information.